gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/ConnectionPool.java

```diff
@@ -59,8 +58,8 @@ final class ConnectionPool {
     private final Client client;
     private final List<Connection> connections;
     private final AtomicInteger open;
+    private final Set<Connection> availableConnections = new CopyOnWriteArraySet<>();
```

Comment:
CopyOnWriteArraySet does have some performance implications for applications which have a lot of insertions and removals although many alternatives carry their own implications. Have you done any profiling which may show whether this is a real concern?

Reply:
I've changed it to ConcurrentLinkedQueue which will be faster for insert and removals and also keep older connections at the head so that they are used next, which is consistent with previous behaviour.